src/main/java/com/socialmedia/service/PostService.java

```diff
@@ -92,6 +94,32 @@ public void updateLikes(Long postId) {
       likes.add(author);
       post.setLikes(likes);
     }
-    update(postId, post);
+    return jpaRepository.save(post);
+  }
+
+  public Post createComment(Long postId, Comment comment) {
+    Principal principal = SecurityContextHolder.getContext().getAuthentication();
+    ApplicationUser author = userService.getById(principal.getName());
+    Post post = getById(postId);
+    comment.setDate(System.currentTimeMillis());
+    comment.setAuthor(author);
+    comment.setPost(post);
+
+    List<Comment> comments = post.getComments();
+    comments.add(comment);
+
+    return jpaRepository.save(post);
   }
+
+  public Post deleteComment(Long postId, Long commentId) {
+    Post post = getById(postId);
```

Comment:
I guess, we need check if it's either comment author, or post author, or post owner to be able to delete comment